indica/workflows/bayes_workflow.py

```diff
@@ -0,0 +1,1065 @@
+
+    @abstractmethod
+    def process_data(self, model_callable: Callable):
+        self.pre_process_data(model_callable)
+        self.opt_data = flatdict.FlatDict(self.binned_data)
+
+
+@dataclass
+class ExpData(DataContext):
+    phantoms = False
+
+    """
+    Considering: either rewriting this class to take over from ReadST40 or vice versa
+    
+    """
+
+    def read_data(self, ):
+        self.reader = ReadST40(self.pulse, tstart=self.tstart, tend=self.tend, dt=self.dt, )
+        self.reader(self.diagnostics, revisions=self.reader_settings.revisions, filters=self.reader_settings.filters)
+        missing_keys = set(self.diagnostics) - set(self.reader.binned_data.keys())
+        if len(missing_keys) > 0:
+            raise ValueError(f"missing data: {missing_keys}")
+        self.equilibrium = self.reader.equilibrium
+        self.transforms = self.reader.transforms
+        # TODO raw data included
+        self.raw_data = self.reader.raw_data
+        self.binned_data = self.reader.binned_data
+
+    def data_strategy(self):
+        raise ValueError("Data strategy: Fail")
+
+    def process_data(self, model_callable: Callable):
+        self.pre_process_data(model_callable)
+        opt_data = flatdict.FlatDict(self.binned_data, ".")
+        if "xrcs.spectra" in opt_data.keys():
+            background = opt_data["xrcs.spectra"].where(
+                (opt_data["xrcs.spectra"].wavelength < 0.392)
+                & (opt_data["xrcs.spectra"].wavelength > 0.388),
+                drop=True,
+            )
+            opt_data["xrcs.spectra"]["error"] = np.sqrt(
+                opt_data["xrcs.spectra"] + background.std(dim="wavelength") ** 2
+            )
+        if "ts.ne" in opt_data.keys():
+            opt_data["ts.ne"]["error"] = opt_data["ts.ne"].max(dim="channel") * 0.05
+
+        if "ts.te" in opt_data.keys():
+            opt_data["ts.ne"]["error"] = opt_data["ts.ne"].max(dim="channel") * 0.05
+        self.opt_data = opt_data
+
+
+@dataclass
+class PhantomData(DataContext):
+    phantoms = True
+
+    def read_data(self, ):
+        self.reader = ReadST40(self.pulse, tstart=self.tstart, tend=self.tend, dt=self.dt, )
+        self.reader(self.diagnostics, revisions=self.reader_settings.revisions, filters=self.reader_settings.filters)
+        missing_keys = set(self.diagnostics) - set(self.reader.binned_data.keys())
+        if len(missing_keys) > 0:
+            raise ValueError(f"missing data: {missing_keys}")
+        self.equilibrium = self.reader.equilibrium
+        self.transforms = self.reader.transforms
+        self.raw_data = {}
+        self.binned_data = {}
+
+    def data_strategy(self):
+        print("Data strategy: Phantom data")
+        return self.model_callable()
+
+    def process_data(self, model_callable: Callable):
+        self.pre_process_data(model_callable)
+        self.opt_data = flatdict.FlatDict(self.binned_data, ".")
+
+
+@dataclass
+class MockData(PhantomData):
```

Comment:
What's the difference between Mock and Phantom? Phantom is taking experimental equilibrium but mock data? Mock is all Mocked??
Also, if we need to abstract (I hear you on this!), is there more we can abstract?